src/data/row.rs

```diff
@@ -41,34 +41,40 @@ impl Row {
     }
 
     pub fn new(column_defs: &[ColumnDef], columns: &[String], values: &[Expr]) -> Result<Self> {
+        if !columns.is_empty() && values.len() != columns.len() {
+            return Err(RowError::DoNotMatchColumnsAndValues.into());
+        }
         if values.len() > column_defs.len() {
```

Comment:
It would be good to use `else if`

Reply:
It seems like a good change. I'll fix it